### ISSUE TITLE
runtime-rs: Cloud Hypervisor now fails with netns error

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#6429]
(... aaaand this is what happens when we have features disabled by default (_mea culpa_ :smile:))

It's looking like #5788 has broken the CH external hypervisor feature (#6201). Enabling CH and testing results in a panic due to the following not being implemented:

```rust
pub(crate) async fn get_vmm_master_tid(&self) -> Result<u32> {
    todo!()
}
```

If I update that code to return the PID of the hypervisor as `Ok(pid)`, runtime-rs crashes as follows:

```bash
ctr: failed to create shim task: Others("failed to handler message create container\n\nCaused by:\n    0: vmm netns guard\n    1: open new netns path /proc/675845/task/675850
/ns/net\n    2: No such file or directory (os error 2)"): unknown
```

Any thoughts @YushuoEdge? Is there anything else the external hypervisors need to do to work with the new API?

**Comment by ananos** [comment]
I think it will definitely break FC. If I understand correctly it is related to this: #6384

**Comment by YushuoEdge** [comment]
Hi @jodh-intel @ananos , this API is added when we need to support OCI hook. Since docker hook(CreateContainer hook) rely on the pid of the container to find out the container namespace on the host. So in runtime-rs + dragonball, we set state.pid to the vmm master thread id for that hook, as it is the only thread in the namespace.

**Comment by jodh-intel** [comment]
Hi @YushuoEdge - Just to confirm, are you planning to raise a PR to resolve this? I can try testing your fix if you haven't tried it with CH?

**Comment by YushuoEdge** [comment]
> Hi @YushuoEdge - Just to confirm, are you planning to raise a PR to resolve this? I can try testing your fix if you haven't tried it with CH?

Yes, I have not tested with CH yet. It seems that CH has not implemented netns related contents when startup, so I am not sure whether it is appropriate to do like the commit. If it's better to discuss under PR, maybe I can raise a fix PR first.

**Comment by YushuoEdge** [comment]
Hi @jodh-intel , I have split the task into two issues: https://github.com/kata-containers/kata-containers/issues/6441, https://github.com/kata-containers/kata-containers/pull/6443. And I have already raised a PR for the second issue, so that your development will not be blocked.

**Comment by jodh-intel** [comment]
Fixed by #6443. Thanks @YushuoEdge ! :smile:
